src/util/bidi.js

```diff
@@ -198,15 +198,16 @@ let bidiOrdering = (function() {
       order.push(new BidiSpan(0, len - m[0].length, len))
     }
 
-    return order
+    order.forDirection = direction
+    return direction == "rtl" ? order.reverse() : order
```

Comment:
Clearing the order on a line direction change would also be trivial. Since we might be generating lots of order arrays, using a cheap representation (plain array) seems a good idea. (Such estimates of how expensive something is are always debatable, in a JIT world, of course.)